qa/L0_backend_python/lifecycle/lifecycle_test.py

```diff
@@ -208,45 +208,6 @@ def test_infer_pymodel_error(self):
                         False, "Wrong exception raised or did not raise an exception"
                     )
 
-    def test_incorrect_execute_return(self):
```

Comment:
qq: why are we removing this test case?

Reply:
This test case expects the model to return `None` or `[None, ...]`, this is an error if returning a response is the only way for non-decoupled models to send a response. With the introduction of response sender for non-decoupled models, the models may return `None` or `[None, ...]` which indicates the response will be sent later after the `execute()` function returns. Thus, the checks on this test case is outdated and therefore removed. The new behavior will be tested by the new response sender tests.

Currently, it is difficult to determine if a model has "forgotten" to send a response and/or final flag, because there is no restriction on how soon the model must send response(s) after `execute()` return, although the destruction of `response_sender` at non-closed state is a hint of such event. However, for the purpose of this test, it is too late for the exception to be returned to the request, because the request is "done" by the time `response_sender` is destructed.

Reply:
Makes sense, thanks for the info!